C/plugins/north/OMF/plugin.cpp

```diff
@@ -812,89 +416,9 @@ void plugin_start(const PLUGIN_HANDLE handle,
 #endif
 
 	Logger* logger = Logger::getLogger();
-	CONNECTOR_INFO* connInfo = (CONNECTOR_INFO *)handle;
-
-	logger->info("Host: %s", connInfo->hostAndPort.c_str());
-	if ((connInfo->PIServerEndpoint == ENDPOINT_OCS) || (connInfo->PIServerEndpoint == ENDPOINT_ADH))
-	{
-		logger->info("Namespace: %s", connInfo->OCSNamespace.c_str());
-	}
-
-	// Parse JSON plugin_data
-	Document JSONData;
-	JSONData.Parse(storedData.c_str());
-	if (JSONData.HasParseError())
-	{
-		logger->error("%s plugin error: failure parsing "
-			      "plugin data JSON object '%s'",
-			      PLUGIN_NAME,
-			      storedData.c_str());
-	}
-	else if (JSONData.HasMember(TYPE_ID_KEY) &&
-		(JSONData[TYPE_ID_KEY].IsString() ||
-		 JSONData[TYPE_ID_KEY].IsNumber()))
-	{
-		// Update type-id in PLUGIN_HANDLE object
-		if (JSONData[TYPE_ID_KEY].IsNumber())
-		{
-			connInfo->typeId = JSONData[TYPE_ID_KEY].GetInt();
-		}
-		else
-		{
-			connInfo->typeId = atol(JSONData[TYPE_ID_KEY].GetString());
-		}
-	}
-
-	// Load sentdataTypes
-	loadSentDataTypes(connInfo, JSONData);
-
-	// Log default type-id
-	if (connInfo->assetsDataTypes.size() == 1 &&
-	    connInfo->assetsDataTypes.find(FAKE_ASSET_KEY) != connInfo->assetsDataTypes.end())
-	{
-		// Only one value: we have the FAKE_ASSET_KEY and no other data
-		Logger::getLogger()->info("%s plugin is using global OMF prefix %s=%d",
-					  PLUGIN_NAME,
-					  TYPE_ID_KEY,
-					  connInfo->typeId);
-	}
-	else
-	{
-		Logger::getLogger()->info("%s plugin is using per asset OMF prefix %s=%d "
-					  "(max value found)",
-					  PLUGIN_NAME,
-					  TYPE_ID_KEY,
-					  getMaxTypeId(connInfo));
-	}
+	OMFInformation *info = (OMFInformation *)handle;
```

Comment:
We had an outside developer a while ago who recommended coding this cast as
`OMFInformation *info = const_cast<OMFInformation *>(handle);`
Certainly not a major problem, or even required. What do you think?

Reply:
It really makes no difference